tests: drivers: dma: remove dead code in test_dma_loop.c

FRASTM opened the pull request:
Fix the error of the [Coverity CID :219489] "Structurally
dead code in tests/drivers/dma/loop_transfer/src/test_dma_loop.c"

Fix https://github.com/zephyrproject-rtos/zephyr/issues/32951

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on tests/drivers/dma/loop_transfer/src/test_dma_loop.c:
```diff
@@ -146,11 +146,10 @@ static int test_loop(void)
 		TC_PRINT("ERROR: unfinished transfer\n");
-		return TC_FAIL;
 		if (dma_stop(dma, chan_id)) {
 			TC_PRINT("ERROR: transfer stop\n");
-			return TC_FAIL;
 		}
```
Is there an interest to keep the "if"?

FRASTM replied:
no, except to report the error on dma_stop

erwango replied:
I don't think there's interest to check behavior after an error

FRASTM commented:
Rebase
Remove the check of  dma_stop in case of error

FRASTM committed:
tests: drivers: dma: remove dead code in test_dma_loop.c

Fix the error of the [Coverity CID :219489] "Structurally
dead code in tests/drivers/dma/loop_transfer/src/test_dma_loop.c"

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented on tests/drivers/dma/loop_transfer/src/test_dma_loop.c:
```diff
@@ -146,11 +146,8 @@ static int test_loop(void)
 	if (transfer_count < TRANSFER_LOOPS) {
 		transfer_count = TRANSFER_LOOPS;
 		TC_PRINT("ERROR: unfinished transfer\n");
+		dma_stop(dma, chan_id);
```
ok, restoring
